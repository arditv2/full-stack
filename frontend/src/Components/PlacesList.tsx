import React, { FC } from 'react';
import { IPlace } from '../App/store/places/types';

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';

interface IProps {
	places: IPlace[];
}

// const PlacesList: FC<IProps> = ({ places }) => {
// 	return (
// 		<Table striped bordered hover className='mt-3'>
// 			<thead>
// 				<tr>
// 					<th>#</th>
// 					<th>Name</th>
// 					<th>Address</th>
// 					<th>Location</th>
// 					<th>Provider</th>
// 				</tr>
// 			</thead>
// 			<tbody>
// 				{places.map((place, index) => (
// 					<tr key={place.id}>
// 						<td id='number'>{index + 1}</td>
// 						<td id='name'>{place.name}</td>
// 						<td id='address'>{place.address}</td>
// 						<td id='location'>
// 							{place.location ? (
// 								<>
// 									<span>
// 										<b>Lat:</b> {place.location.lat}
// 									</span>
// 									<br />
// 									<span>
// 										<b>Long:</b> {place.location.lng}
// 									</span>
// 								</>
// 							) : (
// 								'-'
// 							)}
// 						</td>
// 						<td id='provider'>{place.provider}</td>
// 					</tr>
// 				))}
// 			</tbody>
// 		</Table>
// 	);
// };
const PlacesList: FC<IProps> = ({ places }) => {
	return (
		<Container fluid className='mb-3 p-0'>
			<Row>
				{Boolean(places.length) &&
					places.map((place) => (
						<Col key={place.id} sm='12' md='6' className='mb-3'>
							<Card className='h-100'>
								<Card.Body>
									<Card.Title>{place.name}</Card.Title>
									<Card.Subtitle className='mb-2 text-muted'>
										{place.location && (
											<Card.Text data-test-id='locationText'>
												Lat: {String(place.location.lat).substring(0, 9)} Lng:{' '}
												{String(place.location.lng).substring(0, 9)}
											</Card.Text>
										)}
									</Card.Subtitle>
									<Card.Text>{place.address}</Card.Text>
								</Card.Body>
								<Card.Footer>{place.provider}</Card.Footer>
							</Card>
						</Col>
					))}
			</Row>
		</Container>
	);
};

export default PlacesList;
