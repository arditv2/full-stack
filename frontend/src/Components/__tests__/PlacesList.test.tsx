import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import PlacesList from '../PlacesList';

describe('PlacesList', () => {
	it('renders no cards when places is empty', () => {
		const wrapper: ShallowWrapper = shallow(<PlacesList places={[]} />);
		expect(wrapper.find('Card')).toHaveLength(0);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders a card when places contains one element', () => {
		const wrapper: ShallowWrapper = shallow(
			<PlacesList
				places={[
					{
						id: 'mock-id',
						provider: 'mock-provider',
						name: 'mock-name',
						address: 'mock-address',
						location: { lat: 'mock-lat', lng: 'mock-lng' },
					},
				]}
			/>
		);
		expect(wrapper.find('Card')).toHaveLength(1);
		expect(wrapper.find('CardText[data-test-id="locationText"]')).toHaveLength(
			1
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders a card when places contains one element without location', () => {
		const wrapper: ShallowWrapper = shallow(
			<PlacesList
				places={[
					{
						id: 'mock-id',
						provider: 'mock-provider',
						name: 'mock-name',
						address: 'mock-address',
					},
				]}
			/>
		);
		expect(wrapper.find('Card')).toHaveLength(1);
		expect(wrapper.find('CardText[data-test-id="locationText"]')).toHaveLength(
			0
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders multiple cards when places contains more than one elemen', () => {
		const wrapper: ShallowWrapper = shallow(
			<PlacesList
				places={[
					{
						id: 'mock-id',
						provider: 'mock-provider',
						name: 'mock-name',
						address: 'mock-address',
					},
					{
						id: 'mock-id2',
						provider: 'mock-provider',
						name: 'mock-name2',
						address: 'mock-address2',
					},
				]}
			/>
		);
		expect(wrapper.find('Card')).toHaveLength(2);
		expect(wrapper).toMatchSnapshot();
	});
});
