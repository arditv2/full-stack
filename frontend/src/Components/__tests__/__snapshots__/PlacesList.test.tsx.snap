// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList renders a card when places contains one element 1`] = `
<Container
  className="mb-3 p-0"
  fluid={true}
>
  <Row
    noGutters={false}
  >
    <Col
      className="mb-3"
      key="mock-id"
      md="6"
      sm="12"
    >
      <Card
        body={false}
        className="h-100"
      >
        <CardBody>
          <CardTitle>
            mock-name
          </CardTitle>
          <CardSubtitle
            className="mb-2 text-muted"
          >
            <CardText
              data-test-id="locationText"
            >
              Lat: 
              mock-lat
               Lng:
               
              mock-lng
            </CardText>
          </CardSubtitle>
          <CardText>
            mock-address
          </CardText>
        </CardBody>
        <CardFooter>
          mock-provider
        </CardFooter>
      </Card>
    </Col>
  </Row>
</Container>
`;

exports[`PlacesList renders a card when places contains one element without location 1`] = `
<Container
  className="mb-3 p-0"
  fluid={true}
>
  <Row
    noGutters={false}
  >
    <Col
      className="mb-3"
      key="mock-id"
      md="6"
      sm="12"
    >
      <Card
        body={false}
        className="h-100"
      >
        <CardBody>
          <CardTitle>
            mock-name
          </CardTitle>
          <CardSubtitle
            className="mb-2 text-muted"
          />
          <CardText>
            mock-address
          </CardText>
        </CardBody>
        <CardFooter>
          mock-provider
        </CardFooter>
      </Card>
    </Col>
  </Row>
</Container>
`;

exports[`PlacesList renders multiple cards when places contains more than one elemen 1`] = `
<Container
  className="mb-3 p-0"
  fluid={true}
>
  <Row
    noGutters={false}
  >
    <Col
      className="mb-3"
      key="mock-id"
      md="6"
      sm="12"
    >
      <Card
        body={false}
        className="h-100"
      >
        <CardBody>
          <CardTitle>
            mock-name
          </CardTitle>
          <CardSubtitle
            className="mb-2 text-muted"
          />
          <CardText>
            mock-address
          </CardText>
        </CardBody>
        <CardFooter>
          mock-provider
        </CardFooter>
      </Card>
    </Col>
    <Col
      className="mb-3"
      key="mock-id2"
      md="6"
      sm="12"
    >
      <Card
        body={false}
        className="h-100"
      >
        <CardBody>
          <CardTitle>
            mock-name2
          </CardTitle>
          <CardSubtitle
            className="mb-2 text-muted"
          />
          <CardText>
            mock-address2
          </CardText>
        </CardBody>
        <CardFooter>
          mock-provider
        </CardFooter>
      </Card>
    </Col>
  </Row>
</Container>
`;

exports[`PlacesList renders no cards when places is empty 1`] = `
<Container
  className="mb-3 p-0"
  fluid={true}
>
  <Row
    noGutters={false}
  />
</Container>
`;
